### PR TITLE
[PWN-3467] Added token repository

### PR DESCRIPTION
**Pull request by sidorov-panda**

### CODE REVIEW

File: Sources/SolanaSwift/APIClient/APIClient.swift [hunk 1]
```diff
@@ -381,14 +389,37 @@ extension SolanaAPIClient {
...
-        let result: String? = try await self.get(method: "setLogFilter", params: [filter])
-        return result
+        try await self.get(method: "setLogFilter", params: [filter])
     }
     
     public func validatorExit() async throws -> Bool {
         try await self.get(method: "validatorExit", params: [])
     }
     
+    public func getMultipleMintDatas(mintAddresses: [String], programId: String = PublicKey.tokenProgramId.base58EncodedString) async throws -> [String: Mint] {
```
Comment (TrGiLong) [inline Sources/SolanaSwift/APIClient/APIClient.swift:399]: I think we should keep APIClient much simple and identical to official docs as possible. TokenRepository also share the domain of issue, so it be better if we move them to `TokenRepository/APIClient+Extension.swift`

Comment (sidorov-panda) [reply Sources/SolanaSwift/APIClient/APIClient.swift:399]: Sounds good

File: Sources/SolanaSwift/TokenRepository/TokenRepository.swift [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    // MARK: -
+}
+
+public enum TokensListParserError: Error {
+    case invalidTokenlistURL
+    case invalidTokenlistPath
+}
+
+public class TokensListParser {
```
Comment (TrGiLong) [inline Sources/SolanaSwift/TokenRepository/TokenRepository.swift:111]: Split classes into small files

**Review by bigearsenal — CHANGES_REQUESTED** [review]
1. Please add instruction how to use Token(s)Repository in `README.md` after `APIClient` section
2. Pay attention to naming the classes, variables. The Tokens must be in plural form. For example: TokensRepository (with "s"), not TokenRepository

File: Sources/SolanaSwift/APIClient/APIClient+Extension.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+// MARK: - TokenRepository
+
+public extension SolanaAPIClient {
+    
+    public func getMultipleMintDatas(mintAddresses: [String], programId: String = PublicKey.tokenProgramId.base58EncodedString) async throws -> [String: Mint] {
+        let accounts: [BufferInfo<Mint>] = try await getMultipleAccounts(pubkeys: mintAddresses)
+        var mintDict = [String: Mint]()
+        if accounts.contains(where: { $0.owner != programId }) == true {
+            throw SolanaSDK.Error.other("Invalid mint owner")
```
Comment (bigearsenal) [inline Sources/SolanaSwift/APIClient/APIClient+Extension.swift:11]: Merge branch pwn-3297 and Use SolanaError instead of SolanaSDK.Error

**Review by bigearsenal — CHANGES_REQUESTED** [review]

**Review by bigearsenal — APPROVED** [review]

**Pull request merged**